web/frontend/components/HealthSummary/HealthSummary.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+
+const getHealthIcon = (health) => {
+  switch (health) {
+    case 'critical':
+      return <i className="eos-icons health-summary-icon text-danger">error</i>;
+    case 'warning':
+      return (
+        <i className="eos-icons health-summary-icon text-warning">warning</i>
+      );
+    case 'passing':
+      return (
+        <i className="eos-icons health-summary-icon text-success">
+          check_circle
+        </i>
+      );
+    default:
+      return (
+        <i className="eos-icons health-summary-icon text-muted">
+          fiber_manual_record
+        </i>
+      );
+  }
+};
+
+const any = (predicate, label) =>
+  Object.keys(predicate).reduce((accumulator, key) => {
+    if (accumulator) {
+      return true;
+    }
+    return predicate[key] === label;
+  }, false);
+
+const getCounters = (data) =>
```

Comment:
I think here we should consider to count the `unknowns`. Otherwise, if some summary entry has unknonws and passes, we will compute this pass, which is not entirely sure. I wouldn't count them as pass at least.
But it is true that we don't have any box for them. @abravosuse What should we do here?
- Leave as it is. So if we have some unknown at worst, as pass
- Add other box for the unknown
- Not put other box, but not count neither

Reply:
@arbulu89 for this iteration let's go with the third option: not put other box, but no count neither (I don't want an unknown to be counted as a green). But looking down the road this is how I think it should be:

1. distinguish between stopped and unknown as they are different things (given that grey is the sapcontrol color associated to a stopped status, I suggest we keep grey for stopped and choose another one for unknown. OR, stopped is circle grey and unknown is triangle grey).
2. add as many boxes as possible statuses
3. stablish a clear hegemony among statuses:
I think we all agree that:
- all green > green
- one yellow, rest green > yellow
- one red, rest green/yellow > red
But how stopped or unknown weight in the overall status is something we are going to have to determine component by component. What I mean is that, for example, it's not the same to have an ABAP instance stopped than having the diagnostic agent stopped.
All this to be refined.

Reply:
@dottorblaster I don't know if you read this thread. I think it is missing to be implemented.
We would need to check before the passing, if there is any `unknown`. So simply adding the next should work:

```
      if (any(element, 'unknown')) {
        return { ...accumulator, unknown: accumulator.unknown + 1 };
      }
```
At least, we won't have a new passing counter added in this case (unknowns won't be showed in the frontend, but at least neither false passes)

Reply:
@arbulu89 thanks for the tip, actually I read the thread but I didn't get that detail :sweat_smile: